paparazzi-gradle-plugin/src/main/java/app/cash/paparazzi/gradle/PaparazziPlugin.kt

```diff
@@ -227,6 +244,41 @@ public class PaparazziPlugin @Inject constructor(
 
       recordTaskProvider.configure { it.dependsOn(testTaskProvider) }
       verifyTaskProvider.configure { it.dependsOn(testTaskProvider) }
+      addFailedSnapshotsTaskProvider.configure {
+        it.inputs.dir(buildDirectory.dir("paparazzi/failures"))
+          .withPropertyName("paparazzi.snapshot.failures")
+          .withPathSensitivity(PathSensitivity.RELATIVE)
+        it.inputs.dir(testReportOutputLocation)
+          .withPropertyName("paparazzi.html.report.dir")
+          .withPathSensitivity(PathSensitivity.RELATIVE)
+
+        it.mustRunAfter(testTaskProvider)
+
+        it.doLast {
+          val reportRootDir = testReportOutputLocation.get().asFile
+          val failureSnapshots = File(buildDirectory.asFile.get(), "paparazzi${separator}failures")
+          val generatedSnapshots = failureSnapshots.listFiles() ?: return@doLast
+
+          generatedSnapshots.asSequence().filter { snapshot ->
+            snapshot.name.startsWith("delta")
+          }.forEach { snapshot ->
+            val nameSegments = snapshot.name.split("_")
+            if (nameSegments.size >= 3) {
+              val testClassPackage = nameSegments[0].replace("delta-", "")
+              val testClass = "$testClassPackage.${nameSegments[1]}"
+              val testMethod = nameSegments[2].split(".")[0]
+
+              val testHtmlFile = File(reportRootDir, "classes${separator}$testClass.html")
```

Comment:
Ah, I see what you did here 😺

![Screenshot 2024-08-26 at 1 16 37 AM](https://github.com/user-attachments/assets/6acc6d3c-fcfc-4c1b-ab67-625e678bf7f1)

 ....but copying/manipulating outputs from other Gradle plugins/tasks is inviting maintenance burden 😢 

Ideally, there exists a way to configure the Test Reporting plugin to customize reporting.  If not, then I'd go with @TWiStErRob 's [original suggestion](https://squaremafia.slack.com/archives/C06HF0MLUL8/p1716322051974129?thread_ts=1716321664.541989&cid=C06HF0MLUL8) and working with [PageRenderer](https://cs.android.com/search?q=PageRenderer&sq=&ss=android-studio%2Fplatform%2Ftools%2Fbase), most likely via [TestReport](https://cs.android.com/android-studio/platform/tools/base/+/mirror-goog-studio-main:preview/screenshot/screenshot-test-gradle-plugin/src/main/java/com/android/compose/screenshot/report/TestReport.kt) and [ScreenshotClassPageRenderer](https://cs.android.com/android-studio/platform/tools/base/+/mirror-goog-studio-main:preview/screenshot/screenshot-test-gradle-plugin/src/main/java/com/android/compose/screenshot/report/ScreenshotClassPageRenderer.kt?q=PageRenderer&ss=android-studio%2Fplatform%2Ftools%2Fbase).

Reply:
yep, just took a look at those classes. Unfortunately they are reading the junit xml and creating the entire report themselves which is unfortunate